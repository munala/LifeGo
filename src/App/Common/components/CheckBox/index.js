import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from '@material-ui/core/Checkbox';

import './styles.css';


const CheckBox = ({
  checked,
  style,
  onChange,
  disabled,
}) => (
  <div className="container" >
    <Checkbox
      color="primary"
      onChange={onChange}
      checked={checked}
      disabled={disabled}
      style={style}
    />
  </div>
);

CheckBox.propTypes = {
  checked: PropTypes.bool.isRequired,
  disabled: PropTypes.bool,
  style: PropTypes.shape({}),
  onChange: PropTypes.func,
};

CheckBox.defaultProps = {
  disabled: false,
  style: undefined,
  onChange: () => {},
};

export default CheckBox;
