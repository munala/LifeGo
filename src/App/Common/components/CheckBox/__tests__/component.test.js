import React from 'react';
import { shallow } from 'enzyme';
import CheckBox from '../';

describe('CheckBox tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<CheckBox checked src="source" />);
    expect(wrapper).toMatchSnapshot();
  });
});
