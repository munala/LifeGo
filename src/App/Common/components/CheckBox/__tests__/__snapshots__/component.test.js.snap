// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox tests renders without crashing 1`] = `
<div
  className="container"
>
  <WithStyles(Checkbox)
    checked={true}
    color="primary"
    disabled={false}
    onChange={[Function]}
  />
</div>
`;
